onedal/tests/test_common.py

```diff
@@ -14,27 +14,43 @@
 # limitations under the License.
 # ==============================================================================
 
+import importlib
 import os
 from glob import glob
 
-import pytest
 
-
-def test_sklearn_check_version_ban():
-    """This test blocks the use of sklearn_check_version
-    in onedal files. The versioning should occur in the
-    sklearnex package for clarity and maintainability.
+def _test_primitive_usage_ban(primimtive_name, banned_locations, allowed_locations=None):
```

Comment:
```suggestion
def _check_primitive_usage_ban(primitive_name, package, allowed_locations=None):
```
I know the underscore will avoid pytest including it as a test, but I think changing 'test' to 'check' would be nice.  I think its best also if a single package is checked instead of 'banned_locations'. Also fix a spelling mistake.